apps/web/src/app/student/quiz/_components/QuizView.tsx

```diff
@@ -0,0 +1,107 @@
+} from "@/components/ui/form";
+import { Heading } from "src/components/ui";
+import { CheckCircledIcon } from "@radix-ui/react-icons";
+import { useState } from "react";
+import { Progress } from "@/components/ui/progress";
+
+interface Props {
+  id: number;
+  initialData: TrpcReactQueryOptions["questionBank"]["get"]["initialData"];
+}
+export const QuizView = ({ id, initialData }: Props) => {
+  const { data: questionBank, refetch } = trpc.questionBank.get.useQuery(id, {
+    initialData
+  });
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const question_id = 0; //mock question ID
+  const formSchema = z.object({
+    answer_id: z.coerce.number({
+      required_error: "Please select an answer",
+      invalid_type_error: "Please select an answer"
+    })
+  });
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema)
+  });
+  if (!questionBank) {
+    return <p>Question bank doesnt exist</p>;
+  }
+
+  function onSubmit(values: z.infer<typeof formSchema>) {
+    console.log(values);
+    setIsSubmitted(true);
+  }
+  return (
+    <div className="p-5 flex flex-col h-screen">
+      <Heading>{questionBank.title}</Heading>
+      <p className="text-xl">{questionBank.questions[question_id].title}</p>
+      <Progress className="mt-5" value={33}></Progress>
+      <Form {...form}>
+        <form
+          className="flex flex-col mt-auto mb-10"
+          onSubmit={form.handleSubmit(onSubmit)}
+        >
+          <FormField
+            control={form.control}
+            name="answer_id"
+            render={({ field }) => (
+              <FormItem>
+                <FormControl>
+                  <ButtonGroup
+                    id="form"
+                    {...(isSubmitted && { disabled: true })}
+                    onValueChange={field.onChange}
+                  >
+                    {questionBank.questions[question_id].answers.map(
+                      (answer) => (
+                        <div
+                          key={answer.id}
+                          className="flex items-center justify-center"
+                        >
+                          <ButtonGroupItem
+                            value={String(answer.id)}
+                            className="flex items-center w-full justify-center"
+                          >
+                            {answer.text}
+                            {answer.isCorrect && isSubmitted ? (
+                              <CheckCircledIcon className="ml-2" />
+                            ) : (
+                              false
+                            )}
+                          </ButtonGroupItem>
+                        </div>
+                      )
+                    )}
+                  </ButtonGroup>
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <Button
+            {...(isSubmitted && { disabled: true })}
```

Comment:
change this to 
```
<Button
  disabled={isSubmitted}
  type="submit
.... // rest of props
/>
```
pls do the same above

Reply:
Fixed in 76b261269fe3ba91102d11f0e8059ccf5c56c525